[dynamo] Fakify result of delegate

angelayi opened the pull request:
Summary: Somehow the delegate returns a real tensor result even though we pass in fake tensors. So here we need to convert the result to fake.

Test Plan: `buck2 run @//mode/dev-nosan //on_device_ai/helios/multi_zion:multi_zion_test -- -r test_single_delegate_dsp_only`

Differential Revision: D58617091


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

ydwu4 commented on torch/_inductor/ir.py:
```diff
@@ -5648,9 +5648,10 @@ def create(cls, kernel, *args, **kwargs):
         if example_output is None:
             packed = cls(
-                NoneLayout(None),
+                NoneLayout(device),
```
what's going here for the inductor change?

ydwu4 commented on torch/_dynamo/variables/higher_order_ops.py:
```diff
@@ -1153,6 +1153,7 @@ def call_function(
         )
 
         example_value = lowered_module.original_module.module()(*real_sub_args)
+        example_value = deepcopy_to_fake_tensor(example_value, tx.fake_mode)
```
em...can you try`with tx.fake_mode:` to wrap around the original_module.module() call?

angelayi committed:
[dynamo] Fakify result of delegate (#128752)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/128752

Somehow the delegate returns a real tensor result even though we pass in fake tensors. So here we need to convert the result to fake.

Test Plan: `buck2 run @//mode/dev-nosan //on_device_ai/helios/multi_zion:multi_zion_test -- -r test_single_delegate_dsp_only`

Differential Revision: D58617091
